models/articles.model.js

```diff
@@ -62,12 +81,40 @@ exports.selectArticleComments = (articleId) => {
     WHERE ar.article_id = $1
     ORDER BY com.created_at DESC
     `
-    return db.query(articleQueryString, [articleId]).then((res) => {
-        if (res.rows.length === 0) {
-            return Promise.reject({ status: 404, msg: "article not found" })
-        }
-        return db.query(commentQueryString, [articleId])
-    }).then((res) => {
-        return res.rows
-    })
+    return this.checkArticleExists(articleId)
+        .then(() => {
+            return db.query(commentQueryString, [articleId])
+        })
+        .then((res) => {
+            return res.rows
+        })
+}
+
+exports.insertComment = (articleId, commentBody) => {
+    if (!commentBody.username || !commentBody.body) {
+        return Promise.reject({ status: 400, msg: "Bad Request" })
+    }
+    const { username, body } = commentBody
+    const qStr = `
+        INSERT INTO comments 
+        (body,
+        article_id,
+        author,
+        votes,
+        created_at)
+
+        VALUES
+        ($1,$2,$3,$4,$5)
+        RETURNING *
+        `
+    return this.checkExists("username", "users", username, "user does not exist")
+        .then(() => {
+            return this.checkExists("article_id", "articles", articleId, "article does not exist")
```

Comment:
This works but we have access to SQL errors as the keys referenced are foreign keys. Strongly advise you change this, will give you a call to discuss.